services/datamanager/builtin/file_deletion_test.go

```diff
@@ -0,0 +1,320 @@
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/components/arm"
+	"go.viam.com/rdk/components/gantry"
+	"go.viam.com/rdk/internal/cloud"
+	cloudinject "go.viam.com/rdk/internal/testutils/inject"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/datamanager"
+	"go.viam.com/rdk/services/datamanager/datasync"
+	"go.viam.com/rdk/services/datamanager/internal"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+const (
+	enabledTabularManyCollectorsConfigPath = "services/datamanager/data/fake_robot_with_many_collectors_data_manager.json"
+)
+
+func TestFileDeletionUsageCheck(t *testing.T) {
+	tests := []struct {
+		name              string
+		deletionExpected  bool
+		triggerThreshold  float64
+		captureUsageRatio float64
+		captureDirExists  bool
+	}{
+		{
+			name:              "we should return false from deletion check if not at file system capacity threshold",
+			deletionExpected:  false,
+			triggerThreshold:  .99,
+			captureUsageRatio: .99,
+			captureDirExists:  true,
+		},
+		{
+			name:              "we return true from deletion check if at file system capacity threshold",
+			deletionExpected:  true,
+			triggerThreshold:  math.SmallestNonzeroFloat64,
+			captureUsageRatio: math.SmallestNonzeroFloat64,
+			captureDirExists:  true,
+		},
+		{
+			name: "we return false from deletion check" +
+				"if at file system capacity threshold but not capture dir threshold",
+			deletionExpected:  false,
+			triggerThreshold:  math.SmallestNonzeroFloat64,
+			captureUsageRatio: 1.0,
+			captureDirExists:  true,
+		},
+		{
+			name:              "we should return false from deletion check if capture dir does not exist",
+			deletionExpected:  false,
+			triggerThreshold:  .95,
+			captureUsageRatio: .5,
+			captureDirExists:  false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var tempCaptureDir string
+			if tc.captureDirExists {
+				tempCaptureDir = t.TempDir()
+				// write testing files
+				writeFiles(t, tempCaptureDir, []string{"1.capture", "2.capture"})
+			}
+			// overwrite thresholds
+			fsThresholdToTriggerDeletion = tc.triggerThreshold
+			captureDirToFSUsageRatio = tc.captureUsageRatio
+			logger := logging.NewTestLogger(t)
+			willDelete, err := shouldDeleteBasedOnDiskUsage(context.Background(), tempCaptureDir, logger)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, willDelete, test.ShouldEqual, tc.deletionExpected)
+		})
+	}
+}
+
+func TestFileDeletion(t *testing.T) {
+	tests := []struct {
+		name                  string
+		syncEnabled           bool
+		shouldCancelContext   bool
+		expectedDeletedCount  int
+		fileList              []string
+		syncerInProgressFiles []string
+	}{
+		{
+			name:                 "if sync disabled, file deleter should delete every 4th file",
+			fileList:             []string{"shouldDelete0.capture", "1.capture", "2.capture", "3.capture", "shouldDelete4.capture"},
```

Comment:
Thanks for this naming! Super clear. To make this even more explicit, instead of `expectedDeletedCount`, you can have `expectedDeletedFileNames` and pass in []string{"shouldDelete0.capture", "shouldDelete4.capture"}.